core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2801,19 +2804,11 @@ private JoinBridgeManager<PartitionedLookupSourceFactory> createLookupSourceFact
                     .map(buildSource.getTypes()::get)
                     .collect(toImmutableList());
             List<Type> buildTypes = buildSource.getTypes();
-            JoinBridgeManager<PartitionedLookupSourceFactory> lookupSourceFactoryManager = new JoinBridgeManager<>(
```

Comment:
`JoinBridgeManager` is only needed for grouped execution. We should just remove it https://github.com/trinodb/trino/issues/11936. Then this whole things should simplify and maybe we wouldn't need to extend `LookupSourceFactory` for spilling operator at all.